hilti/toolchain/src/compiler/visitors/validator.cc

```diff
@@ -16,28 +16,33 @@ using util::fmt;
 
 namespace {
 
-struct Visitor : public visitor::PostOrder<void, Visitor> {
+struct VisitorBase {
     // Record error at location of current node.
-    void error(std::string msg, position_t& p, node::ErrorPriority priority = node::ErrorPriority::Normal) {
+    void error(std::string msg, operator_::position_t& p, node::ErrorPriority priority = node::ErrorPriority::Normal) {
```

Comment:
I am not sure whether using an `operator_::position_t` here is always safe (the `position_t` in spicy's `VisitorBase` seems to resolve to the same type).

Can you instead make it accept whatever `position_t` the deriving visitor has, e.g.,

```.cpp
template <typename Position>
void error(std::string msg, Position& p, node::ErrorPriority priority = node::ErrorPriority::Normal) {
```

Here, below for the other methods, and also in spicy's `VisitorBase`.

Reply:
I don't think we need the template, we really just instantiate position with `Node`, I'm changing it to make it explicit through `const visitor::Position<Node&>` (as we do elsewhere).